### PR TITLE
Dynamic Dashboard: Update header view for dashboard screen

### LINKED ISSUE #12511: Card title should not shimmer during loading state
During pull-to-refresh on dashboard, notice how the "Store Setup" title of the old design card remains visible during loading animation. Meanwhile, the new cards add shimmer to the titles, too.

I think it's better if the new cards keep showing the title: it gives better clarity to the users what items are currently loading, and in a sense help them not feel "lost" with the UI change.

https://github.com/woocommerce/woocommerce-ios/assets/266376/a749409b-cbd1-4254-b6b3-2fdbd7bbdec0

### LINKED ISSUE #12517: Update header for the Dashboard screen
image

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #12517 
Also closes #12511 

## Description

This PR adds updates to the dashboard screen UI:
- Set the background color of the navigation bar to be the same as the whole view to match the design.
- Moved the share bar button to the left and changed the edit button title.
- Removed the redacting of the card title when data is being reloaded.
- Reduced the debounce time of the selected chart index to make the chart update faster upon selection.

## Testing instructions

- Enable the `dynamicDashboard` feature flag and build the app.
- Log in to a store and confirm that the dashboard screen header matches the design.
- Enable the Performance card and select any place on the chart (if there is existing data). Confirm that the selection works for both tapping and dragging on the chart.
- Pull-to-refresh the dashboard screen, confirm that the card titles are not redacted.

## Screenshots

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/196">18.3</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Dashboard/StoreStats/StorePerformanceViewModel.swift [hunk 1]
```diff
@@ -475,7 +475,7 @@ private extension StorePerformanceViewModel {
         static let thirtyDaysInSeconds: TimeInterval = 86400*30
 
         /// The wait time before the `StoreStatsUsageTracksEventEmitter.interacted()` is called.
-        static let chartValueSelectedEventsDebounce: TimeInterval = 1.0
+        static let chartValueSelectedEventsDebounce: TimeInterval = 0.5
```
Comment (pmusolino) [inline WooCommerce/Classes/ViewRelated/Dashboard/StoreStats/StorePerformanceViewModel.swift:477]: I believe the new debounce wait time has disrupted the chart's selection and dragging functionalities. The selection is not consistently responsive, and currently, during the dragging of the chart, the following issue is occurring:





This is something I didn't experienced before.

**Review by pmusolino — CHANGES_REQUESTED** [review]
Thank you, Huong, for submitting this PR!

I've reviewed it and noticed a couple of issues. The first issue is detailed in the comments below. The second issue pertains to the card selection functionality. Currently, it is possible to interact with the cards even when they are redacted. To prevent any unintended behavior, I recommend disabling interaction with the cards during the redaction process.

File: WooCommerce/Classes/ViewRelated/Dashboard/StoreStats/StorePerformanceViewModel.swift (hunk 1 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Dashboard/StoreStats/StorePerformanceViewModel.swift:477]: @pmusolino In a previous PR review you [pointed](https://github.com/woocommerce/woocommerce-ios/pull/12483#discussion_r1567980468) out that the constant updates causes the UI to flicker, so I had to debounce the update to minimize that. Now I'm confused about your suggestion in this comment. Should we disable dragging on the chart to simplify the problem? I don't see the point of supporting that.

Comment (pmusolino) [reply WooCommerce/Classes/ViewRelated/Dashboard/StoreStats/StorePerformanceViewModel.swift:477]: After reviewing your [most recent update](https://github.com/woocommerce/woocommerce-ios/commit/e8661b29c9de0e63fe82c5d4b0a7f5c105f58954), I noticed that the dragging functionality was working well for me. However, it appears that reducing the debounce time has introduced this new issue. Previously, the problem manifested as the chart continuously refreshing and displaying data. Now, the cursor moves around the chart but fails to select the interval.

Comment (pmusolino) [reply WooCommerce/Classes/ViewRelated/Dashboard/StoreStats/StorePerformanceViewModel.swift:477]: Btw, as you suggested, I don't see much value in it 👍

Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Dashboard/StoreStats/StorePerformanceViewModel.swift:477]: On a second thought I found a different way to handle the drag gesture on the chart. In b060151553b6b73c0bd39288dc50bf7ad585a441 I updated the selection only after the drag ends, but I still update the chart when the drag changes. LMK what you think!

Simulator Screen Recording - iPhone 15 Pro - 2024-04-22 at 09 57 00

Comment (pmusolino) [reply WooCommerce/Classes/ViewRelated/Dashboard/StoreStats/StorePerformanceViewModel.swift:477]: It works nicely! Thanks Huong 🚢

**Review by pmusolino — APPROVED** [review]
LGTM :shipit:

**Pull request merged**